### PR TITLE
PLAT-61241: Handle direction keys properly over ScrollButtons when `focusableScrollbar` is false

### PR DESCRIPTION
**Pull request by ybsung**
### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

Even though pressing up/down/left/right keys in the ScrollButton when `focusableScrollbar` is false, the ScrollButton keep get focus.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

If there is any spottable component to the direction of the key, give it focus. If not, give a content focus.

The expected result with my PR is...

Step
1. Launch http://nebula.lgsvl.com/enyojs/enact-sampler/develop/
2. Select VirtualList
3. Wheel a little bit

- Hover the previous ScrollButton, then press a 'right' key.
Expected result is that the VirtualList item gets focus.
- Hover the previous ScrollButton, then press a 'left' key.
Expected result is that the VirtualList item gets focus.
- Hover the previous ScrollButton, then press a 'down' key.
Expected result is that the VirtualList item gets focus.
- Hover the previous ScrollButton, then press a 'up' key.
Expected result is that the 'close' button gets focus.
- Hover the previous ScrollButton, then press a 'page down' key.
Expected result is that the VirtualList scrolls.

### Links
[//]: # (Related issues, references)

PLAT-61241

### Comments

Enact-DCO-1.0-Signed-off-by: YB Sung (yb.sung@lge.com)

### CODE REVIEW

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 1]
```diff
@@ -259,16 +269,34 @@ class ScrollButtons extends Component {
...
+				fromNextToPrev = (vertical && direction === 'up') || (!vertical && direction === (rtl ? 'right' : 'left')),
+				fromPrevToNext = (vertical && direction === 'down') || (!vertical && direction === (rtl ? 'left' : 'right'));
+
+			// manually focus the opposite scroll button when 5way pressed
+			if ((fromNextToPrev && target === this.nextButtonNodeRef) ||
+				(fromPrevToNext && target === this.prevButtonNodeRef)) {
+				this.focusOnOppositeScrollButton(ev, direction);
+			}
+		} else {
+			const direction = !vertical && 'up' || rtl && 'right' || 'left';
```
Comment (0x64) [inline packages/moonstone/Scrollable/ScrollButtons.js]: We need a comment about this code for the future works.

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 2]
```diff
@@ -259,16 +269,34 @@ class ScrollButtons extends Component {
...
+
+			if (Spotlight.getPointerMode()) {
+				setTimeout(() => {
```
Comment (0x64) [inline packages/moonstone/Scrollable/ScrollButtons.js]: Let's leave some comment like other async cases.

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 3]
```diff
@@ -279,23 +307,33 @@ class ScrollButtons extends Component {
...
 			{prevButtonDisabled} = this.state,
 			{keyCode} = ev;
 
-		if (isPageUp(keyCode) && !prevButtonDisabled) {
-			Spotlight.focus(this.prevButtonNodeRef);
+		if (isPageUp(keyCode)) {
+			if (focusableScrollbar && !prevButtonDisabled) {
+				Spotlight.focus(this.prevButtonNodeRef);
+			} else if (!focusableScrollbar) {
+				this.onClickPrev(ev);
```
Comment (0x64) [inline packages/moonstone/Scrollable/ScrollButtons.js]: Do we have a requirement to scroll up in this case?

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 4]
```diff
@@ -279,23 +307,33 @@ class ScrollButtons extends Component {
...
 			{nextButtonDisabled} = this.state,
 			{keyCode} = ev;
 
-		if (isPageDown(keyCode) && !nextButtonDisabled) {
-			Spotlight.focus(this.nextButtonNodeRef);
+		if (isPageDown(keyCode)) {
+			if (focusableScrollbar && !nextButtonDisabled) {
+				Spotlight.focus(this.nextButtonNodeRef);
+			} else if (!focusableScrollbar) {
+				this.onClickNext(ev);
```
Comment (0x64) [inline packages/moonstone/Scrollable/ScrollButtons.js]: Do we have a requirement to scroll down in this case?

File: packages/moonstone/Scrollable/ScrollButtons.js (hunk 3 above)
Comment (ybsung) [reply packages/moonstone/Scrollable/ScrollButtons.js]: I just implemented it as Enyo VirtualList and Scroll do.

File: packages/moonstone/Scrollable/ScrollButtons.js (hunk 4 above)
Comment (ybsung) [reply packages/moonstone/Scrollable/ScrollButtons.js]: Same above.

**Comment by 0x64** [comment]
LGTM

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 5]
```diff
@@ -259,16 +269,38 @@ class ScrollButtons extends Component {
...
+				// When changing from "pointer" mode to "5way key" mode,
+				// a pointer is hidden and a last focused item get focused after 30ms.
+				// To make sure the content in `VirtualList` or `Scroller` to be focused after that, we used 50ms.
+				setTimeout(() => {
+					if (Spotlight.getCurrent() === target) {
+						Spotlight.move(direction);
+					}
+				}, 50);
+			} else {
+				Spotlight.move(direction);
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/ScrollButtons.js]: I think we need `if (Spotlight.getCurrent() === target)` this condition.

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 6]
```diff
@@ -66,6 +66,16 @@ class ScrollButtons extends Component {
...
 
+		/**
+		 * When `true`, allows 5-way navigation to the scrollbar controls. By default, 5-way will
+		 * not move focus to the scrollbar controls.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		focusableScrollbar: PropTypes.bool,
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/ScrollButtons.js]: How about just `focusable`?

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 7]
```diff
@@ -279,23 +311,33 @@ class ScrollButtons extends Component {
...
 	onKeyDownPrev = (ev) => {
 		const
+			{focusableScrollbar} = this.props,
 			{nextButtonDisabled} = this.state,
 			{keyCode} = ev;
 
-		if (isPageDown(keyCode) && !nextButtonDisabled) {
-			Spotlight.focus(this.nextButtonNodeRef);
+		if (isPageDown(keyCode)) {
+			if (focusableScrollbar && !nextButtonDisabled) {
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/ScrollButtons.js]: I think `!nextButtonDisabled` should go upper if.

File: packages/moonstone/Scrollable/ScrollButtons.js [hunk 8]
```diff
@@ -279,23 +311,33 @@ class ScrollButtons extends Component {
...
 	onKeyDownNext = (ev) => {
 		const
+			{focusableScrollbar} = this.props,
 			{prevButtonDisabled} = this.state,
 			{keyCode} = ev;
 
-		if (isPageUp(keyCode) && !prevButtonDisabled) {
-			Spotlight.focus(this.prevButtonNodeRef);
+		if (isPageUp(keyCode)) {
+			if (focusableScrollbar && !prevButtonDisabled) {
```
Comment (MikyungKim) [inline packages/moonstone/Scrollable/ScrollButtons.js]: Same here

File: packages/moonstone/Scrollable/ScrollButtons.js (hunk 6 above)
Comment (ybsung) [reply packages/moonstone/Scrollable/ScrollButtons.js]: I'll change it to `focusableScrollButtons`.

**Pull request merged**